common/ledger/blockledger/util.go

```diff
@@ -76,7 +76,7 @@ func CreateNextBlock(rl Reader, messages []*cb.Envelope) *cb.Block {
 	}
 
 	block := protoutil.NewBlock(nextBlockNumber, previousBlockHash)
-	block.Header.DataHash = protoutil.BlockDataHash(data)
+	block.Header.DataHash, _ = protoutil.BlockDataHash(data)
```

Comment:
Here you use the version that verifies the TXs even though it is really unnecessary. In another place below you use the raw computation... I think a uniform approach is better. See comment bellow.